flight/Libraries/math/smoothcontrol.c

```diff
@@ -117,21 +117,31 @@ void smoothcontrol_run(smoothcontrol_state state, uint8_t axis_num, float *new_s
 // Separate handling of thrust.
 void smoothcontrol_run_thrust(smoothcontrol_state state, float *new_signal)
 {
-	/* The -1 on zero throttle causes a huge blip when throttling up, so
+	/* The 0 on zero throttle causes a huge blip when throttling up, so
```

Comment:
I remain concerned about all this stuff.  Need to think about smooth control.

Reply:
It seems to work pretty reliable for me in general. The thrust thing before 3D was a simplistic wrapper because before negative was invalid. Not anymore obviously. Your current changes seem to continue to work as expected for normal flight.